hooks/tk-multi-loader2/tk-3de4_actions.py

```diff
@@ -0,0 +1,289 @@
+        the ``execute_action`` method.
+
+        The ``actions`` is a list of dictionaries holding all the actions to execute.
+        Each entry will have the following values:
+
+            name: Name of the action to execute
+            sg_publish_data: Publish information coming from Shotgun
+            params: Parameters passed down from the generate_actions hook.
+
+        .. note::
+            This is the default entry point for the hook. It reuses the ``execute_action``
+            method for backward compatibility with hooks written for the previous
+            version of the loader.
+
+        .. note::
+            The hook will stop applying the actions on the selection if an error
+            is raised midway through.
+
+        :param list actions: Action dictionaries.
+        """
+        for single_action in actions:
+            name = single_action["name"]
+            sg_publish_data = single_action["sg_publish_data"]
+            params = single_action["params"]
+            self.execute_action(name, params, sg_publish_data)
+
+    def execute_action(self, name, params, sg_publish_data):
+        """
+        Execute a given action. The data sent to this be method will
+        represent one of the actions enumerated by the generate_actions method.
+
+        :param name: Action name string representing one of the items returned by generate_actions.
+        :param params: Params data, as specified by generate_actions.
+        :param sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        :returns: No return value expected.
+        """
+        app = self.parent
+        app.logger.debug(
+            "Execute action called for action %s. "
+            "Parameters: %s. Publish Data: %s" % (name, params, sg_publish_data)
+        )
+
+        # resolve path
+        # toolkit uses utf-8 encoded strings internally and Maya API expects unicode
+        # so convert the path to ensure filenames containing complex characters are supported
+        path = self.get_publish_path(sg_publish_data).decode("utf-8")
+
+        if name == "import_image_seq":
+            self._import_image_seq(path, sg_publish_data)
+
+    ##############################################################################################################
+    # helper methods which can be subclassed in custom hooks to fine tune the behaviour of things
+
+    def _import_image_seq(self, path, sg_publish_data):
+        """
+        Import and image sequence and assign it to the selected cameras.
+
+        :param str path: The file path to load.
+        :param dict sg_publish_data: Shotgun data dictionary with all the standard publish fields.
+        """
+        app = self.parent
+        path, start, end, step = get_hash_path_and_range_info_from_seq(path)
+        name = app.engine.context.entity["name"]
+
+        camera_count = tde4.getNoCameras()
+        if camera_count:
```

Comment:
`camera_count` variable used once, might as well just switch to

```suggestion
        if tde4.getNoCameras():
```